android/quest/src/main/java/org/smartregister/fhircore/quest/ui/main/AppMainActivity.kt

```diff
@@ -104,6 +106,13 @@ open class AppMainActivity : BaseMultiLanguageActivity(), QuestionnaireHandler,
       }
     }
 
+    // Register sync listener then run sync in that order
+    syncListenerManager.registerSyncListener(this, lifecycle)
+
+    CoroutineScope(dispatcherProvider.io()).launch {
```

Comment:
This CoroutineScope is not necessary. The function `fetchNonWorkflowConfigResources` already has a `CoroutineScope` running the code in IO thread. Switching to another context here is not great for performance.

Reply:
Can we call this functional immediately after login before even launching the AppMainActivity? Somewhere in the `QuestLoginService` class?

Reply:
Reason for above the main activity will load the register, and the register has a button for registering client that launches a questionnaire that should have been loaded even when there are no register entries. It would actually be great to display the button only until the download is complete.

Reply:
CC @dubdabasoduba

Reply:
@ellykits This should be loaded immediately after login. Before we start syncing anything else. I am okay with syncing it on the `QuestLoginService`.